test/jdk/java/awt/print/Dialog/DialogCopies.java

```diff
@@ -21,40 +21,71 @@
  * questions.
  */
 
-/**
+/*
  * @test
  * @bug 6357858
  * @summary Job must reports the number of copies set in the dialog.
  * @run main/manual DialogCopies
  */
 
-import java.awt.print.*;
+import java.awt.Frame;
+import java.awt.TextArea;
+import java.awt.BorderLayout;
+import java.awt.print.PrinterJob;
 
 public class DialogCopies {
 
-  static String[] instructions = {
-     "This test assumes and requires that you have a printer installed",
-     "When the dialog appears, increment the number of copies then press OK.",
-     "The test will throw an exception if you fail to do this, since",
-     "it cannot distinguish that from a failure",
-     ""
-  };
+    private static Frame createInstructionUI() {
+        final String instruction = """
+                This test requires that you have a printer. In case if the test
+                system has a virtual printers such as Microsoft Print to PDF or
+                Microsoft XPS Document Writer is installed. Press Cancel
+                button since neither Microsoft Print to PDF or Microsoft XPS
+                Document Writer does not allow setting copies to anything but 1.
+                If printer is installed then PrintDialog is visible increase the
+                number of copies and press OK/Cancel button.""";
```

Comment:
```suggestion
                This test requires that you have a printer.
                
                Press Cancel if your system has only virtual printers such as
                Microsoft Print to PDF or Microsoft XPS Document Writer since
                neither allows setting copies to anything but 1.
                
                If a real printer is installed, select it from the drop-down
                list in the Print dialog and increase the number of copies,
                then press OK button.""";
```

I tried to make the instructions clearer: Press Cancel if … . Otherwise do the following.

Maybe it's better to write _“…since they don't allow setting copies to anything but 1.”_
where “they” refers to any virtual printer.

Reply:
Changed the instruction.